src/app/wallets/get-transactions-for-wallet.ts

```diff
@@ -0,0 +1,71 @@
+import { RepositoryError } from "@domain/errors"
+import { OnChainError, MakeTxFilter, MakeTxDecoder } from "@domain/bitcoin/onchain"
+import { MakeWalletsRepository } from "@services/mongoose"
+import { MakeLedgerService } from "@services/ledger"
+import { MakeOnChainService } from "@services/lnd/onchain-service"
+import { toLiabilitiesAccountId, LedgerError } from "@domain/ledger"
+import { LOOK_BACK } from "@core/utils"
+import { ONCHAIN_MIN_CONFIRMATIONS } from "@config/app"
+import { WalletTransactionHistory } from "@domain/wallets"
+
+// TODO should be exposed via PriceSerivce / LiquidityProvider
+import { getCurrentPrice } from "@services/realtime-price"
+
+export const getTransactionsForWalletId = async ({
+  walletId,
+}: {
+  walletId: WalletId
+}): Promise<{
+  transactions: WalletTransaction[]
+  error?: ApplicationError
```

Comment:
why this method is using a different pattern for return?

Reply:
Because it can return partial results. ie getting ledger transactions works then we want to pass those back to the user but also want to return the error to communicate that we couldn't get the pending txs.

Reply:
uhmm, so if my inner code can return a partial result then should I use this "pattern"? 

you use a lot of wrappers in other parts of the code, why is this better  than use a wrapper and have consistency in the way that we handle errors?

Reply:
> you use a lot of wrappers in other parts of the code, why is this better than use a wrapper and have consistency in the way that we handle errors?

can you show an example here?

Reply:
Just discussing this with @samerbuna.
Think we could standardise on returning an array for partial results:
```
[ResultType, ErrorType]
```

Reply:
```js
export const getTransactionsForWalletId = async ({
  walletId,
}: {
  walletId: WalletId
}): Promise<PartialTransactions | ApplicationError >

// or the name you want
type PartialTransactions = {
  transactions: WalletTransaction[]
  isPartial: boolean
  // or use error 
  error?: ApplicationError

  // later we will have to add something like cursor or other related pagination fields
}
```

yes, probably we should do a log if we have an internal error (getting incoming transactions) but we will not stop the flow of the app so we can simply return Transactions (as alternative) and handle the log/notification internally

Reply:
I think if both a result and an error is returned its clear that it is a partial result.